client/command/screenshot.go

```diff
@@ -0,0 +1,68 @@
+package command
+
+/*
+	Sliver Implant Framework
+	Copyright (C) 2019  Bishop Fox
+
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	sliverpb "github.com/bishopfox/sliver/protobuf/sliver"
+
+	"github.com/desertbit/grumble"
+	"github.com/golang/protobuf/proto"
+)
+
+func screenshot(ctx *grumble.Context, rpc RPCServer) {
+	if ActiveSliver.Sliver == nil {
+		fmt.Printf(Warn + "Please select an active sliver via `use`\n")
+		return
+	}
+
+	if ActiveSliver.Sliver.OS == "darwin" {
+		fmt.Printf(Warn + "Not Implemented\n")
+		return
+	} 
+	
+	fileName := ctx.Flags.String("save")
```

Comment:
We should probably generate the filename ourselves for two reasons:

- it's not clear in which format the screenshot is saved (I tried with `.jpg` and my viewer failed to open it)
- I first thought we needed to pass a directory path, and not a complete file path, which silently failed obviously (my bad for reading the help afterwards).

Something like:
```go
timestamp := time.Now().Format("20060102150405") // format can be something else, see https://golang.org/pkg/time/
fileName := path.Base(fmt.Sprintf("screenshot_%s_%s.png", ActiveSliver.Sliver.Hostname, timestamp)
f, err := ioutil.TempFile("", fileName)
if err != nil {
    // Handle error here
}
...
err := ioutil.WriteFile(f.Name(), screenshotConfigs.Data, 0644)
...
```

Reply:
Third reason: the operator may overwrite an existing file by error.